Solution

kuinorymes opened the pull request.

kuinorymes committed:
Solution

mateacademy-ai-mentor commented on app/main.py:
```diff
@@ -1,8 +1,25 @@
+            if person_dict["name"] == person.name:
+                if "wife" in person_dict:
+                    if person_dict["wife"] is not None:
+                        person.wife = Person.people[person_dict["wife"]]
```
The code attempts to set `person.wife`, but the `Person` class does not have a `wife` attribute defined. Consider adding `self.wife = None` in the `__init__` method of the `Person` class to initialize this attribute.

mateacademy-ai-mentor commented on app/main.py:
```diff
@@ -1,8 +1,25 @@
+                        person.wife = Person.people[person_dict["wife"]]
+                if "husband" in person_dict:
+                    if person_dict["husband"] is not None:
+                        person.husband = Person.people[person_dict["husband"]]
```
Similarly, the code attempts to set `person.husband`, but the `Person` class does not have a `husband` attribute defined. Consider adding `self.husband = None` in the `__init__` method of the `Person` class to initialize this attribute.

mateacademy-ai-mentor reviewed (changes requested):
****

Thank you for your hard work on this Pull Request! 😊 While your code meets most of the task requirements, there are a few critical issues that need to be addressed before we can approve it:

1. **Missing Attributes in `Person` Class**: The `Person` class currently lacks the `wife` and `husband` attributes, which are being set in the code. To resolve this, please add `self.wife = None` and `self.husband = None` in the `__init__` method of the `Person` class to properly initialize these attributes.

2. **Nested `if` Statements**: The code uses nested `if` statements to check for the presence of "wife" and "husband" keys. It's recommended to use logical operators like `and` or `or` to simplify these checks and improve readability.

3. **Use of `dict.get()` Method**: Instead of explicitly checking for the presence of keys "wife" and "husband", consider using the `dict.get()` method. This will make your code cleaner and help avoid potential `KeyError` exceptions.

Once these issues are addressed, your code will be in great shape! Keep up the good work, and don't hesitate to reach out if you have any questions or need further assistance. 🚀

✨ Please rate this review in the Platform's notification center. I will be grateful for your feedback! ✨

🔄 Fix the issues, commit the changes, and then re-request my review using the `Re-request review` button in the Reviewers tab.

kuinorymes committed:
Updated solution

- Used dict.key()
- Removed some if statements
- Add attributes wife & husband to Person

mateacademy-ai-mentor commented on app/main.py:
```diff
@@ -1,8 +1,25 @@
+    for person in person_list:
+        for person_dict in people:
+            if person_dict["name"] == person.name:
+                if person_dict.get("wife") is not None:
+                    person.wife = Person.people[person_dict["wife"]]
+                if person_dict.get("husband") is not None:
+                    person.husband = Person.people[person_dict["husband"]]
```
Consider simplifying the nested `if` statements by using logical operators. For example, you could combine the checks for 'wife' and 'husband' using `if person_dict.get('wife') or person_dict.get('husband'):` and then handle each case within the block. This would improve the readability of your code.

mateacademy-ai-mentor reviewed (approved):
### Review Summary

Great job on addressing the critical issues in your code! 🎉 The `Person` class now correctly includes the `wife` and `husband` attributes, initialized to `None`, which resolves the main issues. You've also implemented the `dict.get()` method effectively to handle potential `KeyError` exceptions. 

However, consider simplifying the nested `if` statements using logical operators to enhance readability. For instance, you could combine the checks for 'wife' and 'husband' using `if person_dict.get('wife') or person_dict.get('husband'):` and manage each case within the block. This small change can make your code cleaner and easier to understand.

Your code meets the task requirements, and the decision to approve is made to help you progress in your learning journey. Keep up the good work, and make sure to review the comments to further improve your coding skills. Remember, addressing these issues will help you avoid similar problems in the future. Keep learning and improving! 🚀

✨ Please rate this review in the Platform's notification center. I will be grateful for your feedback! ✨
